File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int l = 0, r = 1, maxProfit = 0;
        while (r< prices.length){
            if (prices[r]< prices[l]){
                l = r;
            }
            maxProfit = Math.max(maxProfit,prices[r]-prices[l]);
            r++;
        }
        return maxProfit;
    }
}

File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxProfit = 0
        l, r = 0, 1
        # Basically, this is a greedy problem.
        while r < len(prices):
            #profitable transaction
            if prices[r]< prices[l]:
                #buy
                l = r
            if prices[l] < prices[r]:
                #sell
                profit = prices[r] - prices[l]
                maxProfit += profit
                l = r
            r += 1
        # I want to buy all possible profits.

        return maxProfit